filters/include/pcl/filters/impl/filter_indices.hpp

```diff
@@ -73,10 +73,11 @@ pcl::removeNaNFromPointCloud (const pcl::PointCloud<PointT> &cloud_in,
   }
 }
 
-template<typename PointT> void
-pcl::FilterIndices<PointT>::applyFilter (PointCloud &output)
+template<typename PointT, typename DerivedFilter>
+inline void
+pcl::FilterIndices<PointT, DerivedFilter>::applyFilter (PointCloud &output)
```

Comment:
This method should call the new overload with either best_fit or with the inline_executor for the time being.

Reply:
So I don't call the new overload here since it already gets called inside `applyFilter (indices);`
It's not a good idea to call choose the best fit over here, it should be instead delegated to the filter which the user actually uses i.e. the class at the bottom of the inheritance hierarchy in our case currently the `applyFilter (indices)` in Functor Filter chooses the best fit.

Reply:
The two methods are a carbon copy except for the callable (`applyFilter (indices)`, `static_cast<DerivedFilter&>(*this).applyFilter(exec, indices)`). This needs to be refactored

Reply:
Sorry, I didn't quite follow what part needs to be refactored.

Reply:
The following two methods
https://github.com/shrijitsingh99/pcl/blob/51b8f82e152f2a9814dfc193748ddd151c409cac/filters/include/pcl/filters/impl/filter_indices.hpp#L76-L149

Here are the differences between the two
```diff
diff --git a/src1.cpp b/src2.cpp
index a1a5f56..3cc72ca 100644
--- a/src1.cpp
+++ b/src2.cpp
@@ -1,8 +1,15 @@
 template<typename PointT, typename DerivedFilter>
-inline void
-pcl::FilterIndices<PointT, DerivedFilter>::applyFilter (PointCloud &output)
+template <typename Executor> void
+pcl::FilterIndices<PointT, DerivedFilter>::applyFilter (const Executor &exec, PointCloud &output)
 {
-  Indices indices;
+  static_assert(
+      pcl::is_invocable_v<decltype(&DerivedFilter::template applyFilter<Executor>),
+                          DerivedFilter&,
+                          Executor const&,
+                          pcl::Indices&>,
+      "An executor overload for applyFilter doesn't exist.");
+
+  pcl::Indices indices;
   if (keep_organized_)
   {
     if (!extract_removed_indices_)
@@ -10,7 +17,7 @@ pcl::FilterIndices<PointT, DerivedFilter>::applyFilter (PointCloud &output)
       PCL_WARN ("[pcl::FilterIndices<PointT>::applyFilter] extract_removed_indices_ was set to 'true' to keep the point cloud organized.\n");
       extract_removed_indices_ = true;
     }
-    applyFilter (indices);
+    static_cast<DerivedFilter&>(*this).applyFilter(exec, indices);
 
     output = *input_;
 
@@ -27,7 +34,7 @@ pcl::FilterIndices<PointT, DerivedFilter>::applyFilter (PointCloud &output)
   else
   {
     output.is_dense = true;
-    applyFilter (indices);
+    static_cast<DerivedFilter&>(*this).applyFilter(exec, indices);
     pcl::copyPointCloud (*input_, indices, output);
   }
-}
+}

Reply:
I remember asking for this: convert the non-executor into a call for the with-executor function, but then it slipped my eyes. Good catch.

Reply:
Added cbee0c8